src/controller/CHIPDeviceController.h

```diff
@@ -96,6 +98,21 @@ class DLL_EXPORT DevicePairingDelegate
     virtual void OnPairingDeleted(CHIP_ERROR error) {}
 };
 
+class DLL_EXPORT ScriptDevicePairingDelegate final : public DevicePairingDelegate
```

Comment:
could we place this class and implementation into separate h/cpp files?

I would try to avoid adding too many classes and implementations in the same source file.

Reply:
Why isn't this in src/controller/python ?